urllib3/poolmanager.py

```diff
@@ -155,46 +156,53 @@ def clear(self):
         """
         self.pools.clear()
 
-    def connection_from_host(self, host, port=None, scheme='http'):
+    def connection_from_host(self, host, port=None, scheme='http', pool_kwargs=None):
         """
         Get a :class:`ConnectionPool` based on the host, port, and scheme.
 
         If ``port`` isn't given, it will be derived from the ``scheme`` using
-        ``urllib3.connectionpool.port_by_scheme``.
+        ``urllib3.connectionpool.port_by_scheme``. If ``pool_kwargs`` is
+        provided, it is merged with the instance's ``connection_pool_kw``
+        variable and used to create the new connection pool, if one is
+        needed.
         """
 
         if not host:
             raise LocationValueError("No host specified.")
 
-        request_context = self.connection_pool_kw.copy()
+        request_context = self._merge_pool_kwargs(pool_kwargs)
```

Comment:
You're right, it's very confusing. I realized that if we _do_ merge here, though, there's no reason to continue passing the pool_kwargs dictionary around. I've added a new commit that keeps this merge and stops passing the pool_kwargs around. Hopefully that's a clearer way to do this. If not I can back that commit out and mull on it some more :smile:

I regret adding ``connection_from_context`` and ``connection_from_pool_key`` since it only makes it more confusing and I don't think they add a lot of value. Oh well.